### PR TITLE
Fix NullPointerException in ReactShadowNode.toString()

### PR DESCRIPTION
**Pull request by petterh**
Fix `NullPointerException` in `ReactShadowNode.toString`
Simplified `ReactShadowNode.hasNewLayout` since I was already in there

It seems to me unlikely that this bug impacts anything but the debugging experience, so no biggie.

### CODE REVIEW

**Comment by mkonicek** [comment]
Thanks for the fix! When does it happen that `mYogaNode` is null? Do you have exact repro steps, sample code?

**Comment by petterh** [comment]
@mkonicek 
I don't have exact repro steps, but it involved debugging on a physical device that had gone to sleep and woken up again, so probably some activity life cycle thing. I note, however, that many other methods of `ReactShadowNode`  _do_ check for `mYogaNode != null`...

**Comment by janicduplessis** [comment]
Looks like it is null when the node is virtual https://github.com/petterh/react-native/blob/bd92f39cccc66e1b6de3ca8abb826804cd8994b3/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java#L91

Could you also annotate `mYogaNode` with `@Nullable` and check if this produces any other warnings about possible NPE? https://github.com/petterh/react-native/blob/bd92f39cccc66e1b6de3ca8abb826804cd8994b3/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java#L80

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java [hunk 1]
```diff
@@ -770,7 +770,11 @@ public void setMeasureFunction(YogaMeasureFunction measureFunction) {
 
   @Override
   public String toString() {
-    return mYogaNode.toString();
+    if (mYogaNode != null) {
+      return mYogaNode.toString();
+    }
+
+    return getClass().getSimpleName() + " (mYogaNode == null)";
```
Comment (janicduplessis) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java]: maybe write `(virtual node)` instead of `mYogaNode == null`

Comment (petterh) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java]: Done.

**Comment by janicduplessis** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**